hri_safety_sense/src/JoystickHandler.cpp

```diff
@@ -15,93 +15,101 @@
 /**
  * ROS Includes
  */
-#include "ros/ros.h"
-#include "sensor_msgs/Joy.h"
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/msg/joy.hpp"
 
 #include "hri_c_driver/VehicleMessages.h"
-#include "JoystickHandler.h"
+#include "JoystickHandler.hpp"
 
-using namespace hri_safety_sense;
-
-JoystickHandler::JoystickHandler(const std::string &frameId)
+hri_safety_sense::JoystickHandler::JoystickHandler(
+  rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr nodeTopics,
+  rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr nodeLogger,
+  rclcpp::node_interfaces::NodeClockInterface::SharedPtr nodeClock,
+  const std::string &frameId)
 {
-	// Joystick Pub
-	rawLeftPub = rosNode.advertise<sensor_msgs::Joy>("/joy", 10);
-	this->frameId = frameId;
+  // Joystick Pub
+  rawLeftPub = rclcpp::create_publisher<sensor_msgs::msg::Joy>(nodeTopics,
+    "/joy", 10);
+
+  this->nodeLogger = nodeLogger;
+  this->nodeClock = nodeClock;
+  this->frameId = frameId;
 }
 
-JoystickHandler::~JoystickHandler()
+hri_safety_sense::JoystickHandler::~JoystickHandler()
 {
 }
 
-float JoystickHandler::getStickValue(JoystickType joystick)
+float hri_safety_sense::JoystickHandler::getStickValue(JoystickType joystick)
```

Comment:
Since we aren't changing the value of the joystick parameter, we can make this a const reference:

```
float hri_safety_sense::JoystickHandler::getStickValue(const JoystickType & joystick)
```

This avoids copying the JoystickType structure.  Same goes for `getButtonValue` below.

Reply:
d580097